### ISSUE TITLE
Cannot transform compound hunting bow

### ISSUE THREAD

**Issue by Vhari-Maven** [CleverRaven/Cataclysm-DDA#51158]
### Describe the bug

The player should be able to tighten (or loosen) the limbs on the compound hunting bow.  This only works when the item is loaded with an arrow.  Otherwise, the player cannot activate the item (it returns an error stating that it needs a charge).

![Bow Issue 2021-08-28 115119](https://user-images.githubusercontent.com/76548291/131223599-b46d2b39-eab5-4c63-b04a-75e1bc2c67a8.png)

Unfortunately, due to another issue with magazines, the player cannot reload the bow with an arrow.  

### Steps To Reproduce

1. Spawn a compound hunting bow.
2. Attempt to tighten the limbs (it works, because the item starts with an arrow in it).
3. Fire the arrow
4. Attempt to tighten the limbs (it fails, because the item has no charges)

### Expected behavior

The player should be able to tighten (or loosen) the limbs on the compound hunting bow if it is loaded or unloaded.   

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 308d1d2 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]

**Comment by actual-nh** [comment]
What's the bug with "magazines", and has it been reported in an issue previously?

**Comment by Vhari-Maven** [comment]
> What's the bug with "magazines", and has it been reported in an issue previously?

For a while, the player could not reload magazines with loose ammo if the magazine was loaded to the weapon.  As I understand it, this was fixed for most weapons in https://github.com/CleverRaven/Cataclysm-DDA/pull/50281.  However, the player still cannot directly reload bows (at least I can't) with an arrow.  This usually doesn't matter, as the bows could previously only hold one arrow.  

The player can still fire arrows if they have them available (e.g., in their inventory, on the ground, etc.)

I am not an expert, but do not believe that the fix in https://github.com/CleverRaven/Cataclysm-DDA/pull/50281 worked for bows.  I think this is because bows do not have detachable magazines--they just have a built in pocket flagged as a magazine with a size of one.

I have not reported the bow magazine reload issue separately, as I only discovered it when I was having issues adjusting the limbs on the compound bows (as noted in this issue).
